mat64/list_test.go

```diff
@@ -383,7 +384,8 @@ func testTwoInput(c *check.C,
 		&basicMatrix{},
 		&basicVectorer{},
 		&basicSymmetric{},
-		&basicTriangular{},
```

Comment:
For the reason I say below, it can't be that we always have a valid uplo. I think we should leave in the explicit zero value, and then add to the "make rand of" triangular block. Here, if the requested size == 0, the returned triangular has the uplo of the input to makeRandOf. This way we can ensure code works when the receiver has an invalid uplo.